src/core/eme/index.ts

```diff
@@ -60,63 +61,146 @@ function createMediaKeysObs(
   return castToObservable<MediaKeys>(keySystemAccess.createMediaKeys());
 }
 
+/**
+ * Store licence if persistent
+ * @param {Object} keySystem
+ */
+function handleSessionStorage(
+  keySystem: IKeySystemOption
+){
+  if (keySystem.persistentLicense) {
+    if (keySystem.licenseStorage) {
+      $storedSessions.setStorage(keySystem.licenseStorage);
+    } else {
+      const error = new Error("no license storage found for persistent license.");
+      throw new EncryptedMediaError("INVALID_KEY_SYSTEM", error, true);
+    }
+  }
+}
+
+/**
+ * Triggers server certificate setting
+ * @param {ArrayBuffer} serverCertificate
+ * @param {MediaKeys} mediaKeys
+ * @param {ErrorStream} errorStream
+ */
+function setCertificate$(
+  serverCertificate: ArrayBuffer|ArrayBufferView|undefined,
+  mediaKeys: MediaKeys,
+  errorStream: ErrorStream
+) {
+  return (serverCertificate &&
+    typeof mediaKeys.setServerCertificate === "function" ?
+      trySettingServerCertificate(
+        mediaKeys, serverCertificate, errorStream) :
+    Observable.empty()) as Observable<never>; // Typescript hack
+}
```

Comment:
I'm not a fan of a function called ``setCertificate$`` which will not do anything in most cases